### PR TITLE
Fixing bad task output processing

### PR DESCRIPTION
**Pull request by scottfrazer**

### CODE REVIEW

File: src/main/scala/cromwell/engine/backend/Backend.scala [hunk 1]
```diff
@@ -105,7 +107,10 @@ trait Backend {
     def outputAutoConversion(callFqn: String, taskOutput: TaskOutput, rawOutputValue: WdlValue): Try[WdlValue] = {
       rawOutputValue match {
         // Autoconvert String -> File.
-        case v: WdlString if taskOutput.wdlType == WdlFileType => Success(WdlFile(hostAbsoluteFilePath(v.value)))
+        case v: WdlString if taskOutput.wdlType == WdlFileType =>
+          val path = hostAbsoluteFilePath(v.value)
+          if (Files.exists(Paths.get(path))) Success(WdlFile(path))
+          else Failure(new RuntimeException(s"Error attempting to process outputs of call $callFqn:\nCannot process output '${taskOutput.name}' because of error:\nInvalid path: $path"))
```
Comment (mcovarr) [inline src/main/scala/cromwell/engine/backend/Backend.scala]: Super long line!  Something like "expected output file '$yourfile' not found" might be more informative.

File: src/test/scala/cromwell/util/SampleWdl.scala [hunk 2]
```diff
@@ -882,4 +882,24 @@ object SampleWdl {
...
+        |	}
+        |	output {
+        |		# Oops! we made a spelling mistake in our WDL!
+        |		File a = "b"
+        |	}
+        |}
+        |
+        |workflow badExample {
+        |	call bad
+        |}
```
Comment (mcovarr) [inline src/test/scala/cromwell/util/SampleWdl.scala]: Do we have WDL coding standards?  :smile:  I'd advocate for less than 8 chars of indent per level.

File: src/test/scala/cromwell/util/SampleWdl.scala [hunk 3]
```diff
@@ -882,4 +882,24 @@ object SampleWdl {
...
+      """.stripMargin.replaceAll("RUNTIME", runtime)
```
Comment (mcovarr) [inline src/test/scala/cromwell/util/SampleWdl.scala]: don't really need to do the runtime dance if there's not going to be a substitution in any circumstance

**Comment by mcovarr** [comment]
:+1: apart from some of the reformatting you mentioned

File: src/test/scala/cromwell/util/SampleWdl.scala (hunk 2 above)
Comment (scottfrazer) [inline src/test/scala/cromwell/util/SampleWdl.scala]: Actually, the WDL official coding standard calls for 16 spaces per indent level :grin:

**Comment by scottfrazer** [comment]
@mcovarr as first reviewer

**Comment by mcovarr** [comment]
Already :+1: , on to @Horneth

**Comment by cjllanwarne** [comment]
Can I just confirm - this is in Backend not LocalBackend but is checking for a local file path so is the case of a GCS path covered?

File: src/test/scala/cromwell/CromwellTestkitSpec.scala [hunk 4]
```diff
@@ -134,14 +134,23 @@ with DefaultTimeout with ImplicitSender with WordSpecLike with Matchers with Bef
...
         awaitCond(fsm.stateName == WorkflowRunning)
-        awaitCond(fsm.stateName.isTerminal)
-        fsm.stateData should be(WorkflowActor.NoFailureMessage)
+        awaitCond(fsm.stateName == terminalState)
+
+        if (terminalState == WorkflowSucceeded)
+          fsm.stateData should be(WorkflowActor.NoFailureMessage)
+
+        if (terminalState == WorkflowFailed)
+          fsm.stateData.isInstanceOf[WorkflowActor.FailureMessage] shouldEqual true
```
Comment (Horneth) [inline src/test/scala/cromwell/CromwellTestkitSpec.scala]: Any particular reason why the test for `FailureMessage` needs an `isIntanceOf` but not the one for `NoFailureMessage` ?
`fsm.stateData should be(WorkflowActor.NoFailureMessage)` looks better to me.

Comment (scottfrazer) [inline src/test/scala/cromwell/CromwellTestkitSpec.scala]: yeaaaaaaah.... I was hoping nobody noticed that :-D

The reason was that if I tried to do `fsm.stateData should be(WorkflowActor.FailureMessage)`, it wanted me to construct FailureMessage with a String and I just wanted to assert that it failed and I didn't care about the error.

Maybe there's a better way for me express that though.

Comment (Horneth) [inline src/test/scala/cromwell/CromwellTestkitSpec.scala]: Haha ok yeah I guess it's fine then I was just wondering why they were different.
:+1: for me then !

**Comment by scottfrazer** [comment]
@cjllanwarne I know I told you in person, but this code is only executed for the LocalBackend

**Pull request merged**